lib/AST/Builtins.cpp

```diff
@@ -1024,9 +1024,10 @@ static ValueDecl *getAutoDiffApplyAssociatedFunction(
       firstArgGen.build(builder)->castTo<AnyFunctionType>();
   origFnTy = origFnTy->getWithoutDifferentiability()->withExtInfo(
       origFnTy->getExtInfo().withNoEscape(false));
-  auto autodiffBuilder = AutoDiffParameterIndicesBuilder::inferParameters(
-      origFnTy, Context.getStdlibModule());
-  auto *paramIndices = autodiffBuilder.build(Context);
+  AutoDiffParameterIndicesBuilder autoDiffBuilder(origFnTy);
```

Comment:
This isn't the best way to do this, because `AutoDiffIndexSubset` provides a `getDefault()` factory method that is more efficient and easier to use.

```cpp
auto *paramIndices =
    AutoDiffIndexSubset::getDefault(Context, origFnTy->getNumParams(), /*includeAll*/ true);
```

Reply:
`paramIndices` here has type `AutoDiffParameterIndices *`, which is necessary as an argument to `AnyFunctionType::getAutoDiffAssociatedFunctionType`. I don't believe `AutoDiffIndexSubst *` is suitable?

Reply:
Oh shoot, I thought my #25140 had landed.

In this case, use `AutoDiffParameterIndices::get(SmallBitVector(origFnTy->getNumParams(), true), Context)`.

Reply:
Done in https://github.com/apple/swift/pull/25324/commits/3313a81d9762ab44447ca3f170874479d4de5a5e.